sunkit_image/time_lag.py

```diff
@@ -0,0 +1,255 @@
+"""
+This module contains functions for calculating the cross-correlation and time
+lag between intensity cubes.
+"""
+import numpy as np
+
+import astropy.units as u
+
+__all__ = [
+    "cross_correlation",
+    "get_lags",
+    "max_cross_correlation",
+    "time_lag",
+]
+
+
+@u.quantity_input
+def get_lags(time: u.s):
+    """
+    Convert an array of evenly spaced times to an array of time lags
+    evenly spaced between ``-max(time)`` and ``max(time)``.
+    """
+    delta_t = np.diff(time)
+    if not np.allclose(delta_t, delta_t[0]):
+        raise ValueError("Times must be evenly sampled")
+    delta_t = delta_t.cumsum(axis=0)
+    return np.hstack([-delta_t[::-1], np.array([0]), delta_t])
+
+
+@u.quantity_input
+def cross_correlation(signal_a, signal_b, lags: u.s):
+    """
+    Compute cross-correlation between two signals, as a function of lag
+
+    By the convolution theorem the cross-correlation between two signals
+    can be computed as,
+
+    .. math::
+
+        \mathcal{C}_{AB}(\\tau) &= \mathcal{I}_A(t)\star\mathcal{I}_B(t) \\\\
+        &= \mathcal{I}_A(-t)\\ast\mathcal{I}_B(t) \\\\
+        &= \mathscr{F}^{-1}\{\mathscr{F}\{\mathcal{I}_A(-t)\}\mathscr{F}\{\mathcal{I}_B(t)\}\}
+
+    where each signal has been centered and scaled by its mean and standard
+    deviation,
+
+    .. math::
+
+        \mathcal{I}_c(t)=\\frac{I_c(t)-\\bar{I}_c}{\sigma_{c}}
+
+    Additionally, :math:`\mathcal{C}_{AB}` is normalized by the length of
+    the time series.
+
+    Parameters
+    -----------
+    signal_a : array-like
+        The first dimension should correspond to the time dimension
+        and must have length ``(len(lags) + 1)/2``
+    signal_b : array-like
+        Must have the same dimensions as `signal_a`
+    lags : `~astropy.units.Quantity`
+        Evenly spaced time lags corresponding to the time dimension of
+        `signal_a` and `signal_b` running from ``-max(time)`` to
+        ``max(time)``. This is easily constructed using :func:`get_lags`
+
+    Returns
+    -------
+    array-like
+        Cross-correlation as a function of `lags`. The first dimension will be
+        the same as that of `lags` and the subsequent dimensions will be
+        consistent with dimensions of `signal_a` and `signal_b`.
+
+    See Also
+    ---------
+    get_lags
+    time_lag
+    max_cross_correlation
```

Comment:
Will these link like this?

Reply:
They do indeed. They don't need the `sunkit-image.time_lag` in front of them because they are in the same module.